File: week3-exercises02-scala/src/test/scala/com/datastax/workshop/Ex03_Query3a_Insert_Journey.scala
package com.datastax.workshop

import org.junit.jupiter.api.Test
import org.junit.platform.runner.JUnitPlatform
import org.junit.runner.RunWith

@RunWith(classOf[JUnitPlatform])
object Ex03_Query3a_Insert_Journey extends ExerciseBase("Exercise3")

@RunWith(classOf[JUnitPlatform])
class Ex03_Query3a_Insert_Journey {

  import Ex03_Query3a_Insert_Journey._

  @Test
  def insert_a_journey(): Unit = {

    val spaceCraft     = "Crew Dragon Endeavour,SpaceX"
    val journeySummary = "Bring Astronauts to ISS"

    // When inserting a new
    val journeyId = journeyRepo.create(spaceCraft, journeySummary)

    // Validate that journey has been created
    LOGGER.info("Journey created : {}", journeyId)
    LOGGER.info("SUCCESS")

    // Retrieve the Journey just created
    journeyRepo.find(journeyId, spaceCraft) match {
      case None          =>
        LOGGER.info("Could not find Journey by space craft {} with id: {}", spaceCraft, journeyId);
      case Some(journey) =>
        LOGGER.info("Found Journey by space craft {} with id: {}", spaceCraft, journeyId);
        LOGGER.info("Journey: {}", journey);
    }
  }
}

File: week3-exercises01-java/src/test/java/com/datastax/workshop/Ex03_Query3a_Insert_Journey.java
package com.datastax.workshop;

import java.util.Optional;
import java.util.UUID;

import org.junit.jupiter.api.Test;
import org.junit.platform.runner.JUnitPlatform;
import org.junit.runner.RunWith;

@RunWith(JUnitPlatform.class)
public class Ex03_Query3a_Insert_Journey extends ExerciseBase {

    public Ex03_Query3a_Insert_Journey() {
        super("Exercise3");
    }

    @Test
    public void insert_a_journey() {

        String spaceCraft = "Crew Dragon Endeavour,SpaceX";
        String journeySummary = "Bring Astronauts to ISS";

        UUID journeyId = journeyRepo.create(spaceCraft, journeySummary);

        LOGGER.info("Journey created : {}", journeyId);
        LOGGER.info("SUCCESS");

        Optional<Journey> journey = journeyRepo.find(journeyId, spaceCraft);
        if (!journey.isPresent()) {
            LOGGER.info("Could not find Journey by space craft {} with id: {}", spaceCraft, journeyId);
        } else {
            LOGGER.info("Found Journey by space craft {} with id: {}", spaceCraft, journeyId);
            LOGGER.info("Journey: {}", journey.get());
        }
    }
}
